giraffe/src/utils/mosaicTooltip.ts

```diff
@@ -0,0 +1,115 @@
+import {range} from 'd3-array'
+
+import {getRangeLabel} from './tooltip'
+import {X_MIN, X_MAX, FILL, SERIES} from '../constants/columnKeys'
+import {
+  Table,
+  Scale,
+  TooltipData,
+  TooltipColumn,
+  ColumnGroupMap,
+} from '../types'
+
+export const findHoveredBoxes = (
+  boxTable: Table,
+  hoverX: number | null,
+  hoverY: number | null,
+  xScale: Scale<number, number>,
+  yScale: Scale<number, number>,
+  yDomain: number[]
+): number[] => {
+  if (!hoverX || !hoverY) {
+    return []
+  }
+  const xMinData = boxTable.getColumn(X_MIN, 'number')
+  const xMaxData = boxTable.getColumn(X_MAX, 'number')
+  const dataX = xScale.invert(hoverX)
+  const dataY = yScale.invert(hoverY)
+  // Find all boxes whose x extent contain the mouse x position
+  const xIndices = range(0, xMinData.length).filter(
+    i => xMinData[i] <= dataX && xMaxData[i] > dataX
+  )
+
+  const yValMap = new Map()
+  //if series isn't in map yet, add it
+  for (let i = 0; i < yDomain[1]; i++) {
+    const yMin = yScale(i + 1)
+    const yMax = yScale(i)
+    yValMap.set(i, [yMin, yMax])
+  }
+
+  let finalXIndex
+  for (const index of yValMap.keys()) {
+    if (
+      yValMap.get(index)[0] < yScale(dataY) &&
+      yValMap.get(index)[1] >= yScale(dataY)
+    ) {
+      finalXIndex = index
+      break
+    }
+  }
+  let xyIndices
+
+  if (xIndices) {
+    xyIndices = [xIndices[finalXIndex]]
+  } else {
+    xyIndices = []
+  }
+  return xyIndices
+}
+
+export const getMosaicTooltipData = (
+  hoveredBoxRows: number[],
+  boxTable: Table,
+  inputTable: Table,
+  xColKey: string,
+  yColKey: string,
+  fillGroupMap: ColumnGroupMap,
+  fillScale: Scale<number, string>,
+  columnFormatter: (colKey: string) => (x: any) => string
+): TooltipData => {
+  const xMinCol = boxTable.getColumn(X_MIN, 'number')
+  const xMaxCol = boxTable.getColumn(X_MAX, 'number')
+  const valCol = boxTable.getColumn(FILL, 'string')
+  const yCol = boxTable.getColumn(SERIES, 'string')
+  const xFormatter = columnFormatter(xColKey)
+  const yFormatter = columnFormatter(yColKey)
+  const valFormatter = columnFormatter(FILL)
+  const colors = hoveredBoxRows.map(i =>
+    fillScale((valCol[i] as unknown) as number)
+  )
+
+  const xTooltipColumn: TooltipColumn = {
+    key: xColKey,
+    name: inputTable.getColumnName(xColKey),
+    type: inputTable.getColumnType(xColKey),
+    colors,
+    values: hoveredBoxRows.map(i =>
```

Comment:
`i` usually represents an integer counter. something that gets incremented in a loop. I'd like to see this changed to something like `h` or `box` or `row` or `r` or `hbr`. But not `i`, `j`, or `k` cause those are almost universally integer counters.

Reply:
I think that change makes sense. However, the other tooltip files use the variable i for mapping as well. Is it better to change the variable name in this file for clarity or leave it the same as the others for consistency?

Reply:
great question. Either or works. Since there is technical cruft in the form of a poorly named variable elsewhere, we'll want to add a ticket to address that - It's out of the scope of this issue to fix that poorly named variable, but we want to address it, hence the ticket. Since we'll need to add a ticket to address that, and we'll have to do that later, it doesn't really matter all that much if we change this variable now, or later. If we change it now, cool. If we don't change it now, we'll have to also note on the ticket that we need to change the variable name here.